app/models/manageiq/providers/vmware/cloud_manager/event_catcher/stream.rb

```diff
@@ -12,15 +12,11 @@ def self.test_amqp_connection(options = {})
       connection.start
       return true
     rescue Bunny::AuthenticationFailureError => e
-      _log.info("Failed testing rabbit amqp connection: #{e.message}")
-      raise MiqException::MiqInvalidCredentialsError.new "Login failed due to a bad username or password."
-    rescue Bunny::TCPConnectionFailedForAllHosts => e
-      raise MiqException::MiqHostError.new "Socket error: #{e.message}"
+      raise MiqException::MiqInvalidCredentialsError.new "bad username or password for #{options[:hostname]}"
+    rescue Bunny::TCPConnectionFailed, Bunny::TCPConnectionFailedForAllHosts
+      raise MiqException::MiqHostError.new "cannot reach #{options[:hostname]}:#{options[:port]}"
     rescue
-      _log.info("Failed testing rabbit amqp connection for #{options[:hostname]}. ")
       raise
-    ensure
-      connection.close if connection.respond_to? :close
```

Comment:
@miha-plesko I wonder if we can get into a state where this wouldn't work properly, namely an exception would occur, but the connection would be open somehow. Would this additional condition in the deleted line 23 help resolving your case: 

```
connection.close if connection.respond_to? :close and connection.open?
```

From [Bunny doc](http://www.rubydoc.info/github/ruby-amqp/bunny/Bunny/Session#open%3F-instance_method).